src/Diagnostics.DataProviders/ArmClient.cs

```diff
@@ -7,22 +7,24 @@ namespace Diagnostics.DataProviders
     internal class ArmClient : IGeoMasterClient
     {
         private const string CsmEndpointUrl = "https://management.azure.com/";
-        public HttpClient Client { get; }
-        public Uri BaseUri { get; }
+        private static readonly HttpClient _client = new HttpClient();
+        public HttpClient Client => _client;
+        public Uri BaseUri => _client.BaseAddress;
+        public string AuthenticationToken { get; }
 
-        public ArmClient(GeoMasterDataProviderConfiguration configuration)
+        static ArmClient()
         {
-            var handler = new HttpClientHandler();
-            BaseUri = new Uri(CsmEndpointUrl);
+            _client.BaseAddress = new Uri(CsmEndpointUrl);
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
 
-            Client = new HttpClient(handler)
+        public ArmClient(GeoMasterDataProviderConfiguration configuration)
+        {
+            if (configuration.Token == null)
             {
-                BaseAddress = BaseUri
-            };
-
-            Client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            Client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", configuration.Token);
-            Client.BaseAddress = BaseUri;
+                throw new ArgumentNullException("NULL TOKEN!!");
```

Comment:
Can we change this to `ArgumentNullException(nameof(configuration.Token))`?

Reply:
Thanks. I tried it and it showed "Value cannot be null.Parameter name: Token" instead of "configuration.Token." So I will put a literal string "configuration.Token" instead.